### ISSUE TITLE
Review and update all gulp tasks

### ISSUE THREAD

**Issue by JacobDB** [JacobDB/new-site#151]
I've gotten a lot better at wiring JavaScript since I wrote these, and it's time I go through them and clean them up. Some general ideas:

- [x] Linting no longer needs to be it's own step in a task; this was originally set up this way in order to exclude vendor files, but vendor files are now handled by webpack.

- [x] Allow setting upload protocol without having to use different flags; prompt for sftp, ftps, or ftp with the `--upload` flag and adapt accordingly.

- [x] Don't load BrowserSync at all unless `--sync` is passed.

- [x] Switch from `gulp-file-inculde` to `lodash.template` -- https://www.npmjs.com/package/lodash.template

- [x] Remove code removal feature from init task.

- [x] Replace `eyeglass` with `postcss-import` and `postcss-inline-svg`.

- [x] Only set up the current endpoints connection information (i.e. if the endpoint protocol is `ftp`, don't set up `sftp` stuff in the `upload` task).

**Comment by JacobDB** [comment]
> Linting no longer needs to be it's own step in a task; this was originally set up this way in order to exclude vendor files, but vendor files are now handled by webpack.

This is technically done, but the `scripts` task doesn't mark itself as ran correctly. Can't figure out why, see [StackOverflow](https://stackoverflow.com/questions/52613825/merged-gulp-tasks-never-fire-end-event).

**Comment by JacobDB** [comment]
> Delete the entire dist folder if --dist is passed, rather than specific sub folders.
> 
> Delete the entire dev folder if --force is passed.

Doing this seems to force the entire task sequence to always run. Not sure why, but not a big enough deal for me to worry about. Removing these from the list.

**Comment by JacobDB** [comment]
Fixed the FTP settings thing &ndash; I broke up the `.ftpconfig` in to three separate files: `.ftpconfig,` `.ftpspecific`, and `.sftpspecific`. `.ftpconfig` contains all the shared parameters between the two, while the other two contain specific parameters for each protocol. During the prompt process, I pull down `.ftpconfig`, prompt for the protocol, and the download the remaining parameters from the matching file and concatenate them in to the `.ftpconfig` endpoint object. Works perfectly now :)
